src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -16,15 +19,23 @@ trait AuthResource extends RestResource {
 
   def authenticated(request: HttpServletRequest)(fn: Identity => Response): Response = {
     val requestWrapper = new RequestFacade(request)
-    val maybeIdentity = result(authenticator.authenticate(requestWrapper))
-    maybeIdentity.map { identity =>
-      try {
-        fn(identity)
-      } catch {
-        case e: AccessDeniedException => withResponseFacade(authorizer.handleNotAuthorized(identity, _))
-      }
-    }.getOrElse {
-      withResponseFacade(authenticator.handleNotAuthenticated(requestWrapper, _))
+    val authenticationRequest = authenticator.authenticate(requestWrapper)
+
+    Await.ready(authenticationRequest, config.zkTimeoutDuration)
+
+    authenticationRequest.value.get match {
```

Comment:
I know... If you do an Await.result instead (and do the try/catch) you can
avoid the .value.get. It's unfortunate that there's no Await.maybe that
returns a Try
On Tue, Jun 28, 2016 at 2:35 PM Gastón Kleiman <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/api/AuthResource.scala
> <https://github.com/mesosphere/marathon/pull/4036#discussion_r68747452>:
>
> > @@ -16,15 +19,23 @@ trait AuthResource extends RestResource {
> >
> >    def authenticated(request: HttpServletRequest)(fn: Identity => Response): Response = {
> >      val requestWrapper = new RequestFacade(request)
> > -    val maybeIdentity = result(authenticator.authenticate(requestWrapper))
> > -    maybeIdentity.map { identity =>
> > -      try {
> > -        fn(identity)
> > -      } catch {
> > -        case e: AccessDeniedException => withResponseFacade(authorizer.handleNotAuthorized(identity, _))
> > -      }
> > -    }.getOrElse {
> > -      withResponseFacade(authenticator.handleNotAuthenticated(requestWrapper, _))
> > +    val authenticationRequest = authenticator.authenticate(requestWrapper)
> > +
> > +    Await.ready(authenticationRequest, config.zkTimeoutDuration)
> > +
> > +    authenticationRequest.value.get match {
>
> I used scala.utils.Try instead, to avoid catching the exceptions thrown
> by fn.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4036/files/68f4e6d526ddec9f6a5aed3031272096948afdc7#r68747452>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAweZl9nA5vhbWqd-8mgeEEfRYtO-Ai3ks5qQRUWgaJpZM4I_-h3>
> .
>